cognite/neat/_rules/transformers/_converters.py

```diff
@@ -233,95 +234,167 @@ class ToExtension(RulesTransformer[DMSRules, DMSRules]):
     def __init__(
         self,
         new_model_id: DataModelIdentifier,
-        org_name: str | None = None,
-        mode: Literal["extension", "solution"] = "extension",
+        org_name: str = "My",
+        type_: Literal["enterprise", "solution"] = "enterprise",
+        mode: Literal["read", "write"] = "read",
+        dummy_property: str = "GUID",
     ):
         self.new_model_id = DataModelId.load(new_model_id)
         if not self.new_model_id.version:
-            raise ValueError("Version is required for the new model.")
+            raise NeatValueError("Version is required for the new model.")
 
         self.org_name = org_name
         self.mode = mode
+        self.type_ = type_
+        self.dummy_property = dummy_property
 
     def transform(self, rules: DMSRules | OutRules[DMSRules]) -> JustRules[DMSRules]:
         # Copy to ensure immutability
-
         reference_model = self._to_rules(rules)
-        if self.org_name is None and reference_model.metadata.as_data_model_id() in COGNITE_MODELS:
-            raise NeatValueError(
-                f"Organization name is required when mapping to {reference_model.metadata.as_data_model_id()}"
-            )
         reference_model_id = reference_model.metadata.as_data_model_id()
 
-        dump = reference_model.dump()
+        # if model is solution then we need to get correct space for views and containers
+        if self.type_ == "solution":
+            if self.mode not in ["read", "write"]:
+                raise NeatValueError(f"Unsupported mode: {self.mode}")
+
+            if reference_model_id in COGNITE_MODELS:
+                print(
+                    "You are building solution model on top of"
+                    " Cognite Data Model, this is not recommended."
+                    " Build the solution model on top of enterprise model instead."
+                )
```

Comment:
Should probably raise a warning here, and then do some magic in neat session to catch it.